lib/main/permissions/permission_service.dart

```diff
@@ -0,0 +1,17 @@
+import 'package:permission_handler/permission_handler.dart';
+
+class PermissionService {
+  Future<bool> isGranted(Permission permission) async {
+    final current = await permission.status;
+    switch (current) {
+      case PermissionStatus.granted:
+        return true;
+      case PermissionStatus.denied:
```

Comment:
how about the case of `isPermanentlyDenied`?

Reply:
If the status is `PermissionStatus.permanentlyDenied`, no request can be done to revert it, and user has to go to system settings to change it. So it is default case, which return `false`.